simulation/revolve2/simulation/custom_vector/object/vector2.py

```diff
@@ -0,0 +1,206 @@
+"""A representation of a 2d Vector."""
+from __future__ import annotations
+
+from numbers import Number
+from typing import Any
+
+import numpy as np
+from pyrr.objects.base import BaseMatrix33, BaseVector, NpProxy
+
+from .. import vector2
+
+
+class BaseVector2(BaseVector):  # type:ignore
```

Comment:
What is the reason this class exist. why does vector2 not directly inherit from BaseVector?